julia/DataJoint2Julia/src/datajointJulia/connection.py

```diff
@@ -0,0 +1,298 @@
+"""
+This module contains the Connection class that manages the connection to the database,
+ and the `conn` function that provides access to a persistent connection in datajoint.
+"""
+import warnings
+from contextlib import contextmanager
+import pymysql as client
+import logging
+from getpass import getpass
+
+from datajoint.settings import config
+from datajoint import errors
+from datajoint.dependencies import Dependencies
+
+# client errors to catch
+client_errors = (client.err.InterfaceError, client.err.DatabaseError)
+
+
+def translate_query_error(client_error, query):
```

Comment:
Why does existing functionality from `datajoint-python` have to be replicated here?